addons/website/static/src/js/widgets/media.js

```diff
@@ -1,7 +1,22 @@
 odoo.define('website.widgets.media', function (require) {
 'use strict';
 
-const {ImageWidget} = require('wysiwyg.widgets.media');
+const {
+    FileWidget,
+    ImageWidget,
+} = require('wysiwyg.widgets.media');
+
+FileWidget.include({
+    _afterSave() {
+        // When website editor is used, image_changed needs to be awaited
+        // to perform the automatic optimization of the image. To achieve
+        // this, the event is also triggered from Wysiwyg's openMediaDialog.
+        if (!document.querySelector("#oe_snippets.o_loaded")) {
+            return this._super(...arguments);
+        }
+        // else do not call super
+    },
+});
```

Comment:
Nothing here is code specific to website so it could be directly added in web_editor (actually, is not it required for mass_mailing?).

However, it's kinda ugly to check that with a querySelector. Can't it be with an option passed to the media dialog?